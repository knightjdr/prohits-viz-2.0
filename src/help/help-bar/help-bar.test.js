import React from 'react';
import { shallow } from 'enzyme';

import HelpBar from './help-bar';

const showPanel = jest.fn();

describe('HelpBar', () => {
  test('It renders', () => {
    const wrapper = shallow(
      <HelpBar
        isPanelVisible={false}
        isSmallScreen={false}
        showPanel={showPanel}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('It renders with panel', () => {
    const wrapper = shallow(
      <HelpBar
        isPanelVisible
        isSmallScreen={false}
        showPanel={showPanel}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('It renders on small screen', () => {
    const wrapper = shallow(
      <HelpBar
        isPanelVisible={false}
        isSmallScreen
        showPanel={showPanel}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('It renders on small screen with panel', () => {
    const wrapper = shallow(
      <HelpBar
        isPanelVisible
        isSmallScreen
        showPanel={showPanel}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
