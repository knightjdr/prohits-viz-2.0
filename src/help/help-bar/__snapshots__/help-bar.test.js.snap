// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelpBar It renders 1`] = `
<div
  className="HelpBar-container"
>
  <div
    className="HelpBar-panel boxshadow "
  >
    <div
      className="HelpBar-breadcrumb "
    >
      <withRouter(TreeRoutesContainer) />
    </div>
  </div>
  <Button
    className="HelpBar-button "
    ghost={false}
    loading={false}
    onClick={[Function]}
    prefixCls="ant-btn"
    shape="circle"
    type="primary"
  >
    <FontAwesomeIcon$1
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            448,
            512,
            Array [],
            "f329",
            "M448 80v352c0 26.51-21.49 48-48 48H48c-26.51 0-48-21.49-48-48V80c0-26.51 21.49-48 48-48h352c26.51 0 48 21.49 48 48zM240.971 361.941l135.515-135.515c9.373-9.373 9.373-24.569 0-33.941l-16.971-16.971c-9.373-9.373-24.569-9.373-33.941 0L224 277.088 122.427 175.515c-9.373-9.373-24.569-9.373-33.941 0l-16.971 16.971c-9.373 9.373-9.373 24.569 0 33.941L207.03 361.942c9.372 9.372 24.568 9.372 33.941-.001z",
          ],
          "iconName": "chevron-square-down",
          "prefix": "fas",
        }
      }
      listItem={false}
      mask={null}
      name=""
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      transform={
        Object {
          "rotate": 270,
        }
      }
    />
  </Button>
</div>
`;

exports[`HelpBar It renders on small screen 1`] = `
<div
  className="HelpBar-container"
>
  <div
    className="HelpBar-panel boxshadow HelpBar-panel-transition-out"
  >
    <div
      className="HelpBar-breadcrumb HelpBar-breadcrumb-transition-out"
    >
      <withRouter(TreeRoutesContainer) />
    </div>
  </div>
  <Button
    className="HelpBar-button HelpBar-button-transition-out"
    ghost={false}
    loading={false}
    onClick={[Function]}
    prefixCls="ant-btn"
    shape="circle"
    type="primary"
  >
    <FontAwesomeIcon$1
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            448,
            512,
            Array [],
            "f329",
            "M448 80v352c0 26.51-21.49 48-48 48H48c-26.51 0-48-21.49-48-48V80c0-26.51 21.49-48 48-48h352c26.51 0 48 21.49 48 48zM240.971 361.941l135.515-135.515c9.373-9.373 9.373-24.569 0-33.941l-16.971-16.971c-9.373-9.373-24.569-9.373-33.941 0L224 277.088 122.427 175.515c-9.373-9.373-24.569-9.373-33.941 0l-16.971 16.971c-9.373 9.373-9.373 24.569 0 33.941L207.03 361.942c9.372 9.372 24.568 9.372 33.941-.001z",
          ],
          "iconName": "chevron-square-down",
          "prefix": "fas",
        }
      }
      listItem={false}
      mask={null}
      name=""
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      transform={
        Object {
          "rotate": 270,
        }
      }
    />
  </Button>
</div>
`;

exports[`HelpBar It renders on small screen with panel 1`] = `
<div
  className="HelpBar-container"
>
  <div
    className="HelpBar-panel boxshadow HelpBar-panel-transition-in"
  >
    <div
      className="HelpBar-breadcrumb HelpBar-breadcrumb-transition-in"
    >
      <withRouter(TreeRoutesContainer) />
    </div>
  </div>
  <Button
    className="HelpBar-button HelpBar-button-transition-in"
    ghost={false}
    loading={false}
    onClick={[Function]}
    prefixCls="ant-btn"
    shape="circle"
    type="primary"
  >
    <FontAwesomeIcon$1
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            448,
            512,
            Array [],
            "f329",
            "M448 80v352c0 26.51-21.49 48-48 48H48c-26.51 0-48-21.49-48-48V80c0-26.51 21.49-48 48-48h352c26.51 0 48 21.49 48 48zM240.971 361.941l135.515-135.515c9.373-9.373 9.373-24.569 0-33.941l-16.971-16.971c-9.373-9.373-24.569-9.373-33.941 0L224 277.088 122.427 175.515c-9.373-9.373-24.569-9.373-33.941 0l-16.971 16.971c-9.373 9.373-9.373 24.569 0 33.941L207.03 361.942c9.372 9.372 24.568 9.372 33.941-.001z",
          ],
          "iconName": "chevron-square-down",
          "prefix": "fas",
        }
      }
      listItem={false}
      mask={null}
      name=""
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      transform={
        Object {
          "rotate": 90,
        }
      }
    />
  </Button>
</div>
`;

exports[`HelpBar It renders with panel 1`] = `
<div
  className="HelpBar-container"
>
  <div
    className="HelpBar-panel boxshadow "
  >
    <div
      className="HelpBar-breadcrumb "
    >
      <withRouter(TreeRoutesContainer) />
    </div>
  </div>
  <Button
    className="HelpBar-button "
    ghost={false}
    loading={false}
    onClick={[Function]}
    prefixCls="ant-btn"
    shape="circle"
    type="primary"
  >
    <FontAwesomeIcon$1
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            448,
            512,
            Array [],
            "f329",
            "M448 80v352c0 26.51-21.49 48-48 48H48c-26.51 0-48-21.49-48-48V80c0-26.51 21.49-48 48-48h352c26.51 0 48 21.49 48 48zM240.971 361.941l135.515-135.515c9.373-9.373 9.373-24.569 0-33.941l-16.971-16.971c-9.373-9.373-24.569-9.373-33.941 0L224 277.088 122.427 175.515c-9.373-9.373-24.569-9.373-33.941 0l-16.971 16.971c-9.373 9.373-9.373 24.569 0 33.941L207.03 361.942c9.372 9.372 24.568 9.372 33.941-.001z",
          ],
          "iconName": "chevron-square-down",
          "prefix": "fas",
        }
      }
      listItem={false}
      mask={null}
      name=""
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      transform={
        Object {
          "rotate": 90,
        }
      }
    />
  </Button>
</div>
`;
